sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -58,10 +58,13 @@ private string GetSessionFilePath()
             TestContext.TestAdapter testAdapter = TestContext.CurrentContext.Test;
 
             string name = new string(testAdapter.Name.Select(c => s_invalidChars.Contains(c) ? '%' : c).ToArray());
+            string directorySuffix = testAdapter.Properties.ContainsKey(ClientTestFixtureAttribute.RecordingDirectoryKey) ?
+                testAdapter.Properties.Get(ClientTestFixtureAttribute.RecordingDirectoryKey).ToString() :
+                string.Empty;
 
             string className = testAdapter.ClassName.Substring(testAdapter.ClassName.LastIndexOf('.') + 1);
             string fileName = name + (IsAsync ? "Async" : string.Empty) + ".json";
-            return Path.Combine(TestContext.CurrentContext.TestDirectory, "SessionRecords", className, fileName);
+            return Path.Combine(TestContext.CurrentContext.TestDirectory, "SessionRecords", className, directorySuffix, fileName);
```

Comment:
What do you think about following the pattern we have for classe:

`TestTypeName(parameter)/TestMethod(parameters..)` ?

Reply:
For example: 
`\SessionRecords\TableClientLiveTests(parameter)\myTest(CosmosTable)\myTest.json`

or

`\SessionRecords\TableClientLiveTests(parameter)\myTest(CosmosTable).json`
?

Reply:
Take the following recording as example: 

https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/tests/SessionRecords/BlobBaseClientTests/DownloadFileAsync_Parallel_LargeBlobs(1073741824%2C1).json

The test itself is parametrized so we are getting the `(1073741824,1)` in the test name. What I'm talking about is:

`SessionRecords/BlobBaseClientTests(CosmosTable)/DownloadFileAsync_Parallel_LargeBlobs(1073741824,1).json`

Reply:
ah, makes sense. Thanks for the clarification. 
Yeah, I like that.